### PR TITLE
Valhalla ideas

### PR DESCRIPTION
**Pull request by InfernoVulpix**
A bunch of first-step implementations and a few new ideas in Valhalla.

### CODE REVIEW

**Comment by Cogito** [comment]
Hey awesome!

First question, is there any chance you can split into individual commits, where each commit introduces a specific thing? Would make reviewing a little easier, but if not will look at it all.

Splitting may also allow for specific parts to be merged while others are being polished.

Reading through the changes now.

File: loops/actionList.js [hunk 1]
```diff
@@ -3458,8 +3468,307 @@ Action.AcceptDonations = new Action("Accept Donations", {
...
+		addResource("gold", -100);
+	},
+	visible() {
+		return towns[4].getLevel("Tour") >= 70;
+	},
+	unlocked() {
+		return towns[4].getLevel("Tour") >= 90;
+	},
+	finish() {
+		addResource("steed", true);
```
Comment (Cogito) [inline loops/actionList.js:3768]:
```suggestion
		addResource("wingedSteed", true);
```

File: loops/actionList.js [hunk 2]
```diff
@@ -3458,8 +3468,307 @@ Action.AcceptDonations = new Action("Accept Donations", {
...
+    unlocked() {
+        return true;
+    },
+    finish() {
+        addMana(resources.gold * 50);
+        resetResource("gold");
+    },
+});
+
+Action.SellArtifacts = new Action("Sell Artifacts", {
```
Comment (Cogito) [inline loops/actionList.js:3499]:
```suggestion
Action.SellArtifact = new Action("Sell Artifact", {
```

File: loops/img/giftArtifacts.svg [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     showgrid="false"
+     inkscape:zoom="0.995913"
+     inkscape:cx="113.51421"
+     inkscape:cy="221.44574"
+     inkscape:window-x="-9"
+     inkscape:window-y="-9"
+     inkscape:window-maximized="1"
+     inkscape:current-layer="Layer_1" />
+  <title
+     id="title2">takeArtifacts</title>
```
Comment (Cogito) [inline loops/img/giftArtifacts.svg:50]:
```suggestion
     id="title2">giftArtifact</title>
```

- also suggest changing the filename to the singular, though I don't think I can do that through this interface.

File: loops/img/giftArtifacts.svg [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+   xmlns:svg="http://www.w3.org/2000/svg"
+   xmlns="http://www.w3.org/2000/svg"
+   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   id="Layer_1"
+   data-name="Layer 1"
+   viewBox="0 0 474.37902 473.93699"
+   version="1.1"
+   sodipodi:docname="sellArtifacts.svg"
```
Comment (Cogito) [inline loops/img/giftArtifacts.svg:14]:
```suggestion
   sodipodi:docname="giftArtifact.svg"
```

File: loops/img/sellArtifacts.svg [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+   xmlns:svg="http://www.w3.org/2000/svg"
+   xmlns="http://www.w3.org/2000/svg"
+   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   id="Layer_1"
+   data-name="Layer 1"
+   viewBox="0 0 474.37902 473.93699"
+   version="1.1"
+   sodipodi:docname="sellArtifacts.svg"
```
Comment (Cogito) [inline loops/img/sellArtifacts.svg:14]:
```suggestion
   sodipodi:docname="sellArtifact.svg"
```

- again filename would need to be changed here as well

File: loops/img/sellArtifacts.svg [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+     showgrid="false"
+     inkscape:zoom="0.995913"
+     inkscape:cx="113.51421"
+     inkscape:cy="221.44574"
+     inkscape:window-x="-9"
+     inkscape:window-y="-9"
+     inkscape:window-maximized="1"
+     inkscape:current-layer="Layer_1" />
+  <title
+     id="title2">takeArtifacts</title>
```
Comment (Cogito) [inline loops/img/sellArtifacts.svg:50]:
```suggestion
     id="title2">sellArtifact</title>
```

File: loops/lang/en-EN/game.xml [hunk 7]
```diff
@@ -1694,35 +1709,70 @@
...
             <!--TODO: give this action story-->
         </accept_donations>
+		<get_mana>
+            <label>Get Mana</label>
+            <tooltip><![CDATA[
+            1 gold = 50 mana. Buys all the mana you can.
+        ]]></tooltip>
+            <!--TODO: give this action story-->
+		</get_mana>
+		<sell_artifacts>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1719]:
```suggestion
		<sell_artifact>
```

File: loops/lang/en-EN/game.xml [hunk 8]
```diff
@@ -1694,35 +1709,70 @@
...
+			<label>Sell Artifacts</label>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1720]:
```suggestion
			<label>Sell Artifact</label>
```

File: loops/lang/en-EN/game.xml [hunk 9]
```diff
@@ -1694,35 +1709,70 @@
...
+			<tooltip><![CDATA[
+			Sell the artifacts you found on Mt. Olympus as priceless ancient heirlooms.<br>Costs one artifact.<br>Gives 50 gold.
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1722]:
```suggestion
			Sell one of the artifacts you found on Mt. Olympus as a priceless ancient heirloom.<br>Costs one artifact.<br>Gives 50 gold.
```

File: loops/lang/en-EN/game.xml [hunk 10]
```diff
@@ -1694,35 +1709,70 @@
...
+		]]></tooltip>
+            <!--TODO: give this action story-->
+		</sell_artifacts>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1725]:
```suggestion
		</sell_artifact>
```

File: loops/lang/en-EN/game.xml [hunk 11]
```diff
@@ -1694,35 +1709,70 @@
...
+		<gift_artifacts>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1726]:
```suggestion
		<gift_artifact>
```

File: loops/lang/en-EN/game.xml [hunk 12]
```diff
@@ -1694,35 +1709,70 @@
...
+			<label>Gift Artifacts</label>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1727]:
```suggestion
			<label>Gift Artifact</label>
```

File: loops/lang/en-EN/game.xml [hunk 13]
```diff
@@ -1694,35 +1709,70 @@
...
+			<tooltip><![CDATA[
+			Generously give the precious artifacts away.  It's okay, with each act of generosity you earn friends in high places.<br>Costs one artifact.<br>Gives one favor.
+		]]></tooltip>
+            <!--TODO: give this action story-->
+		</gift_artifacts>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1732]:
```suggestion
		</gift_artifact>
```

File: loops/lang/en-EN/game.xml [hunk 14]
```diff
@@ -1694,35 +1709,70 @@
...
         <mercantilism>
             <label>Mercantilism</label>
             <tooltip><![CDATA[
             Practice your merchandising skills, which are sorely lacking considering how poor of a deal you're getting from those mana merchants.
         ]]></tooltip>
             <!--TODO: give this action story-->
         </mercantilism>
+		<enchant_gear>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1740]:
```suggestion
		<enchant_armor>
```

File: loops/lang/en-EN/game.xml [hunk 15]
```diff
@@ -1694,35 +1709,70 @@
...
+			<label>Enchant Gear</label>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1741]:
```suggestion
			<label>Enchant Armor</label>
```

File: loops/lang/en-EN/game.xml [hunk 16]
```diff
@@ -1694,35 +1709,70 @@
...
+			<tooltip><![CDATA[
+			Pull some strings to convince a divine blacksmith to enchant your armor.<br>Costs one favor and one armor.<br>Enchanted armor is three times as effective as normal armor.
+		]]></tooltip>
+            <!--TODO: give this action story-->
+		</enchant_gear>
```
Comment (Cogito) [inline loops/lang/en-EN/game.xml:1746]:
```suggestion
		</enchant_armor>
```

File: loops/img/wingedSteed.svg [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+   xmlns:svg="http://www.w3.org/2000/svg"
+   xmlns="http://www.w3.org/2000/svg"
+   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   id="Layer_1"
+   data-name="Layer 1"
+   viewBox="0 0 474.37902 473.93699"
+   version="1.1"
+   sodipodi:docname="sellArtifacts.svg"
```
Comment (Cogito) [inline loops/img/wingedSteed.svg:14]:
```suggestion
   sodipodi:docname="wingedSteed.svg"
```

File: loops/img/wingedSteed.svg [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+     showgrid="false"
+     inkscape:zoom="0.995913"
+     inkscape:cx="113.51421"
+     inkscape:cy="221.44574"
+     inkscape:window-x="-9"
+     inkscape:window-y="-9"
+     inkscape:window-maximized="1"
+     inkscape:current-layer="Layer_1" />
+  <title
+     id="title2">takeArtifacts</title>
```
Comment (Cogito) [inline loops/img/wingedSteed.svg:50]:
```suggestion
     id="title2">wingedSteed</title>
```

File: loops/img/enchantGear.svg [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+   xmlns:svg="http://www.w3.org/2000/svg"
+   xmlns="http://www.w3.org/2000/svg"
+   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   id="Layer_1"
+   data-name="Layer 1"
+   viewBox="0 0 474.37902 473.93699"
+   version="1.1"
+   sodipodi:docname="sellArtifacts.svg"
```
Comment (Cogito) [inline loops/img/enchantGear.svg:14]:
```suggestion
   sodipodi:docname="enchantArmor.svg"
```

File: loops/img/enchantGear.svg [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+     showgrid="false"
+     inkscape:zoom="0.995913"
+     inkscape:cx="113.51421"
+     inkscape:cy="221.44574"
+     inkscape:window-x="-9"
+     inkscape:window-y="-9"
+     inkscape:window-maximized="1"
+     inkscape:current-layer="Layer_1" />
+  <title
+     id="title2">takeArtifacts</title>
```
Comment (Cogito) [inline loops/img/enchantGear.svg:50]:
```suggestion
     id="title2">enchantArmor</title>
```

**Comment by InfernoVulpix** [comment]
> First question, is there any chance you can split into individual commits, where each commit introduces a specific thing? Would make reviewing a little easier, but if not will look at it all.
> 
> Splitting may also allow for specific parts to be merged while others are being polished.
I'm looking into how to rebase, not sure yet how to do it properly but once I figure it out I'll split things up.

**Comment by Cogito** [comment]
> > First question, is there any chance you can split into individual commits, where each commit introduces a specific thing? Would make reviewing a little easier, but if not will look at it all.
> > Splitting may also allow for specific parts to be merged while others are being polished.
> 
> I'm looking into how to rebase, not sure yet how to do it properly but once I figure it out I'll split things up.

The way I would do this is to, on a new branch, update your working tree to be the same as with all the changes (so the tip of your current master) but then reset the branch pointer to point to omsi:master.

Then, add only those changes you want in each commit at a time, commit them, and repeat.

**Comment by InfernoVulpix** [comment]
@omsi6, as this is a pull request on your repository, github administration says I need your permission for it to be deleted.  If you can approve this then they can delete this pull request and we can move forward with the other pull request.

**Comment by omsi6** [comment]
How do I give permission to delete it? If you just require a comment from me saying it's okay, then here is a comment where I say it's fine to delete this PR.

**Comment by edbrannin** [comment]
@InfernoVulpix Are you trying to push a new or rebased branch with the same name?

If so, it might be easiest to push the update as a new branch with a different name.

**Comment by InfernoVulpix** [comment]
I already have a new pull request set up, I just need this one closed because some of the commits on this one used my real name and I want to keep them separate.  And now, since everything in this pull request (including the suggestions in the comments provided by Cogito) has been integrated into the new pull request, the only thing left to do is to have this pull request deleted so my real name is no longer visible.

**Pull request closed without merging**